src/drivers/imx/interrupt.c

```diff
@@ -3,26 +3,437 @@
+ * Interrupts 448-479 at offset 1
+ * Interrupts 416-447 at offset 2
+ * ...
+ * Interrupts 64-95 at offset 13
+ * Interrupts 32-63 at offset 14
+ * Interrupts 0-31 at offset 15
+ *
+ * The IRQSTR_CH_* macros perform the second part of this calculation
+ * (offset) automatically.
+ */
+
+#define IRQSTR_CHANCTL			0x00
+#define IRQSTR_CH_MASK(n)		(0x04 + 0x04 * (15 - (n)))
+#define IRQSTR_CH_SET(n)		(0x44 + 0x04 * (15 - (n)))
+#define IRQSTR_CH_STATUS(n)		(0x84 + 0x04 * (15 - (n)))
+#define IRQSTR_MASTER_DISABLE		0xC4
+#define IRQSTR_MASTER_STATUS		0xC8
+
+#define IRQSTR_INT_REG(irq)		((irq) / 32)
+#define IRQSTR_INT_BIT(irq)		((irq) % 32)
+#define IRQSTR_INT_MASK(irq)		(1 << IRQSTR_INT_BIT(irq))
+
+#define IRQSTR_RESERVED_IRQS_NUM	32
+#define IRQSTR_IRQS_NUM			512
+#define IRQSTR_IRQS_REGISTERS_NUM	16
+
+/* HW register access helper methods */
+
+static inline void irqstr_write(uint32_t reg, uint32_t value)
+{
+	io_reg_write(IRQSTR_BASE_ADDR + reg, value);
+}
+
+static inline uint32_t irqstr_read(uint32_t reg)
+{
+	return io_reg_read(IRQSTR_BASE_ADDR + reg);
+}
+
+static inline void irqstr_update_bits(uint32_t reg, uint32_t mask,
+				      uint32_t value)
+{
+	io_reg_update_bits(IRQSTR_BASE_ADDR + reg, mask, value);
+}
+
+/* IRQ_STEER helper methods
+ * These methods are usable in any IRQ_STEER driver, not specific to SOF
+ */
+
+static void irqstr_enable_hw(void)
+{
+	irqstr_write(IRQSTR_CHANCTL, 1);
+}
+
+static void irqstr_disable_hw(void)
+{
+	irqstr_write(IRQSTR_CHANCTL, 0);
+}
+
+/* irqstr_get_status_word() - Get an interrupt status word
+ * @index The index of the status word
+ *
+ * Get the status of interrupts 32*index .. 32*(index+1)-1 in a word.
+ * This status is in one hardware register.
+ * Return: Status register word for the corresponding interrupts
+ */
+static uint32_t irqstr_get_status_word(uint32_t index)
+{
+	/* First 32 interrupts are reserved, we just give 0 */
+	if (!index)
+		return 0;
+	/* On out of range for our platform, be silent */
+	if (index > 15)
+		return 0;
+	return irqstr_read(IRQSTR_CH_STATUS(index));
+}
+
+/* Mask, that is, disable interrupts */
+static void irqstr_mask_int(uint32_t irq)
+{
+	uint32_t mask;
+
+	if (irq < IRQSTR_RESERVED_IRQS_NUM || irq >= IRQSTR_IRQS_NUM)
+		return; // Unusable interrupts
+	mask = IRQSTR_INT_MASK(irq);
+	irqstr_update_bits(IRQSTR_CH_MASK(IRQSTR_INT_REG(irq)), mask, 0);
+}
+
+/* Unmask, that is, enable interrupts */
+static void irqstr_unmask_int(uint32_t irq)
+{
+	uint32_t mask;
+
+	if (irq < IRQSTR_RESERVED_IRQS_NUM || irq >= IRQSTR_IRQS_NUM)
+		return; // Unusable interrupts
+	mask = IRQSTR_INT_MASK(irq);
+	irqstr_update_bits(IRQSTR_CH_MASK(IRQSTR_INT_REG(irq)), mask, mask);
+}
+
+__attribute__((unused))
+static void irqstr_mask_group(uint32_t index)
+{
+	uint32_t mask;
+
+	/* Disables output interrupt line index */
+	if (index >= 8)
+		return;
+	mask = 1 << index;
+	irqstr_update_bits(IRQSTR_MASTER_DISABLE, mask, mask);
+}
+
+__attribute__((unused))
```

Comment:
Why do you add these if they are unused?

Reply:
Would the alternative be with #if 0? Or should I just remove these?

Reply:
@paulstelian97 just remove them. @jajanusz we thought on adding them as some sort of documentation, but I to remove confusion the best way is to remove the functions and only add them if they are used.

Reply:
Alright, done.